killbill-integration-tests/core/usage/test_capacity.rb

```diff
@@ -168,7 +178,78 @@ def test_multiple_units
 
       check_invoice_no_balance(second_invoice, 5.00, 'USD', '2015-03-01')
       check_usage_invoice_item(second_invoice.items[0], second_invoice.invoice_id, 5.00, 'USD', 'USAGE', 'basic-monthly', 'basic-monthly-evergreen', 'basic-monthly-usage1', '2015-02-01', '2015-03-01')
+      check_invoice_item_detail(second_invoice.items[0],
+                                [{:unit_type => 'bandwith-meg-sec', :unit_qty => 101, :unit_price => 5.0 },
+                                 {:unit_type => 'members', :unit_qty => 10, :unit_price => 5.0 }], 5.0)
+    end
+
+    def test_multiple_units_detail_mode
+      detail_mode
+
+      bp = create_entitlement('basic-monthly')
+
+      # Both unit are part of tier 1
+      usage_input = [{:unit_type => 'members',
+                      :usage_records => [{:record_date => '2015-01-01', :amount => 6},
+                                         {:record_date => '2015-01-07', :amount => 3},
+                                         {:record_date => '2015-01-15', :amount => 10},
+                                         {:record_date => '2015-01-23', :amount => 8}]
+                     },
+                     {:unit_type => 'bandwith-meg-sec',
+                      :usage_records => [{:record_date => '2015-01-02', :amount => 100},
+                                         {:record_date => '2015-01-07', :amount => 3},
+                                         {:record_date => '2015-01-15', :amount => 10},
+                                         {:record_date => '2015-01-25', :amount => 50}]
+                     }]
+
+      record_usage(bp.subscription_id, usage_input, @user, @options)
+
+      kb_clock_add_months(1, nil, @options)
+      wait_for_expected_clause(1, @account, @options, &@proc_account_invoices_nb)
+
+      all_invoices = @account.invoices(true, @options)
+      sort_invoices!(all_invoices)
+      assert_equal(1, all_invoices.size)
+      first_invoice = all_invoices[0]
 
+      check_invoice_no_balance(first_invoice, 1.00, 'USD', '2015-02-01')
+      assert_equal(2, first_invoice.items.size)
+      check_invoice_item_w_quantity(first_invoice.items[0], first_invoice.invoice_id, 0.5, 'USD', 'USAGE', 'basic-monthly', 'basic-monthly-evergreen',
```

Comment:
For `CAPACITY`, both `detail` and `aggregate` modes will actually output the same invoice line items, we cannot _distribute_ the amount that way. The reason for it is because your overall consumption tells you which tier you end up paying based on limits (unlike `CONSUMABLE` mode where the customer is invoiced on a per unit basis).

See expected output as described by https://github.com/killbill/killbill/issues/839#issuecomment-362474537 (http://killbill.io/blog/usage-billing/ might help with some background too).